### PR TITLE
neighbour-finding CLI to clize

### PR DESCRIPTION
**Pull request by arh89**
Testing:
 - [x] Ran tests and they passed OK

### CODE REVIEW

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4691129&bid=15684335)

File: lib/improver/cli/neighbour_finding.py [hunk 1]
```diff
@@ -31,146 +31,25 @@
...
-def process(orography, landmask, site_list,
-            all_methods=False, land_constraint=None, minimum_dz=None,
-            search_radius=None, node_limit=None, site_coordinate_system=None,
-            site_coordinate_options=None, site_x_coordinate=None,
+from improver import cli
+
+
+@cli.clizefy
+@cli.with_output
+def process(orography: cli.inputcube,
```
Comment (benfitzpatrick) [inline lib/improver/cli/neighbour_finding.py]: Codacy Issue found: [Too many local variables (33/20)](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4691129)

File: lib/improver/tests/acceptance/test_neighbour_finding.py [hunk 2]
```diff
@@ -84,8 +88,11 @@ def test_nearest_minimum_dz(tmp_path, domain, model):
     orography_path = kgo_dir / f"inputs/{model}_orography.nc"
     landmask_path = kgo_dir / f"inputs/{model}_landmask.nc"
     output_path = tmp_path / "output.nc"
-    args = [sites_path, orography_path, landmask_path, output_path,
-            "--minimum_dz", "--search_radius", "50000"]
+    args = [orography_path, landmask_path,
+            "--site-list", sites_path,
+            "--minimum-dz",
+             "--search-radius", "50000",
```
Comment (bayliffe) [inline lib/improver/tests/acceptance/test_neighbour_finding.py]: Bonus extra space indentation.
```suggestion
            "--search-radius", "50000",
```

File: lib/improver/tests/acceptance/test_neighbour_finding.py [hunk 3]
```diff
@@ -143,11 +155,13 @@ def test_alternative_coordinate_system(tmp_path):
...
-            "--site_coordinate_options", coord_opts_json,
-            "--site_x_coordinate", "projection_x_coordinate",
-            "--site_y_coordinate", "projection_y_coordinate"]
+    args = [orography_path, landmask_path,
+            "--site-list", sites_path,
+            "--site-coordinate-system", "LambertAzimuthalEqualArea",
+            "--site-coordinate-options", coord_opts_json,
+            "--site-x-coordinate", "projection_x_coordinate",
+            "--site-y-coordinate", "projection_y_coordinate",
+            "--output", output_path]
```
Comment (bayliffe) [inline lib/improver/tests/acceptance/test_neighbour_finding.py]: The test data relating to this was fixed, but this test has not been updated. Please remove the xfail decorator and use a standard compare at the bottom:
```
acc.compare(output_path, kgo_path)
```

Comment (bayliffe) [reply lib/improver/tests/acceptance/test_neighbour_finding.py]: I'll do this in a separate little PR.

**Pull request merged**